torch/csrc/cuda/Module.cpp

```diff
@@ -149,7 +149,8 @@ PyObject * THCPModule_getRNGState(PyObject *_unused)
 PyObject * THCPModule_setRNGState(PyObject *_unused, PyObject *obj)
 {
   HANDLE_TH_ERRORS
-  if (!THPVariable_Check(obj) || THPVariable_UnpackData(obj).type().ID() != at::TypeID::CPUByte) {
+  auto& data_type = THPVariable_Unpack(obj).type();
```

Comment:
This is completely unsafe and can lead to memory errors. You haven't even checked that `obj` is a variable!

Reply:
Fixed in https://github.com/pytorch/pytorch/pull/17474.